sympy/assumptions/ask_generated.py

```diff
@@ -99,27 +99,29 @@ def get_known_facts_cnf():
         Q.complex | ~Q.imaginary,
         Q.complex | ~Q.real,
         Q.complex | ~Q.transcendental,
+        Q.extended_real | ~Q.infinite,
```

Comment:
Why has this file changed?

Reply:
It seems that change of class structure affects the order of this dictionary. `Q.infinite` was `UndefinedPredicate` before migration, but now it's `InfinitePredicate`.

Reply:
As an aside in the old assumptions we do not have this rule and it is inconsistent with zoo.

It is possible to be both infinite and not extended real e.g.:
```python
In [33]: x = Symbol('x', infinite=True, extended_real=False)

In [34]: x.is_infinite
Out[34]: True

In [35]: x.is_extended_real
Out[35]: False

In [36]: zoo.is_infinite
Out[36]: True

In [37]: zoo.is_extended_real
Out[37]: False
```
Not for this PR but that should be fixed.